distribution/pull_v2.go

```diff
@@ -367,13 +369,34 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named, platform
 			"remote": ref,
 		}))
 
-	manifest, err := p.manifestStore.Get(ctx, specs.Descriptor{
+	desc := specs.Descriptor{
 		MediaType: mt,
 		Digest:    dgst,
 		Size:      size,
-	})
+	}
+	manifest, err := p.manifestStore.Get(ctx, desc)
 	if err != nil {
-		return false, err
+		if isTagged && isNotFound(errors.Cause(err)) {
+			logrus.WithField("ref", ref).WithError(err).Debug("Falling back to pull manifest by tag")
```

Comment:
Happy to change message to whatever...

Reply:
I would actually leave a message saying "Non conforming registry detected, falling back to pull manifest by tag" as it gives the reason, and people might notice it more.